src/software/thunderscope/robot_communication.py

```diff
@@ -81,27 +99,36 @@ def run(self):
         that the robots timeout and stop.
 
         NOTE: If disconnect_fullsystem_from_robots is called, then the packets
-        will not be forwarded to the robots. 
+        will not be forwarded to the robots.
 
         send_override_primitive_set can be used to send a primitive set, which
         is useful to dip in and out of robot diagnostics.
 
         """
         while True:
-            if self.fullsystem_connected_to_robots:
+            # total primitives for all robots
+            robot_primitives = dict()
```

Comment:
is there an advantage to declaring it this way?

Reply:
Not really, it's marginally faster (https://stackoverflow.com/questions/17097985/dict-vs-in-python-which-is-better) 

From what I can tell, the rest of the codebase uses `{}` so it might be better to stay consistent